### PR TITLE
falcon API on Google App Engine

### PR DESCRIPTION
**Pull request by archelogos**
A quick tutorial that shows how to create the basic components of a Python API made with Falcon.
It uses the App Engine Standard Environment

### CODE REVIEW

**Comment by jmdobry** [comment]
Thanks for your submission!

LGTM, leaving it up to @steveperry-53 for final approval.

**Comment by JimTravis** [comment]
Steve is out of the office. Reassigned to chenopis.

File: tutorials/appengine-python-falcon/index.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+                def on_patch(self, req, resp):
+                    ...
+
+                def on_delete(self, req, resp):
+                    ...
+
+    3. In the [`middleware.py`][middleware] file you can find the `AuthMiddleware` class
+    which is used to ensure that all requests are authenticated.
+    Because this is just an example, it is not implemented any kind
```
Comment (chenopis) [inline tutorials/appengine-python-falcon/index.md]: "...not implemented any kind..." => "...not implemented **with** any kind..."

File: tutorials/appengine-python-falcon/index.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                        token = req.get_header('Authorization')
+                        ...
+
+    4. The [`hooks.py`][hooks] file contains definitions of custom functions that can be called
+    before or after an endpoint function is executed. They can be used, for instance, to validate
+    the input data or serialize the API responses.
+
+## Running the app
+
+1. Install the dependencies into the `lib` folder with pip
```
Comment (chenopis) [inline tutorials/appengine-python-falcon/index.md]: Add a period to the end of the sentence.

File: tutorials/appengine-python-falcon/index.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+        pip install -t lib -r requirements.txt
+
+2. Execute the following command to run the app
```
Comment (chenopis) [inline tutorials/appengine-python-falcon/index.md]: Add a period to the end of the sentence.

File: tutorials/appengine-python-falcon/index.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+        gcloud app deploy
+
+2. Visit `http://[YOUR_PROJECT_ID].appspot.com` to see the deployed app.
+
+    Replace `[YOUR_PROJECT_ID]` with your Google Cloud Platform project ID.
+
+3. Run the following command to view your app
+
+    gcloud app browse
```
Comment (chenopis) [inline tutorials/appengine-python-falcon/index.md]: Add additional indent for code format.

File: tutorials/appengine-python-falcon/index.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+3. Deploy the Python app to Google App Engine standard environment.
+
+## Costs
+
+This tutorial does not use billable components of Google Cloud Platform so
+you do not need to enable the billing for your project to complete this tutorial.
+
+## Before you begin
+
+1.  Create a project in the [Google Cloud Platform Console](https://console.cloud.google.com/).
```
Comment (chenopis) [inline tutorials/appengine-python-falcon/index.md]: Suggestion: "Create a project in the [Google Cloud Platform Console](https://console.cloud.google.com/), **and make note of the project ID**."

File: tutorials/appengine-python-falcon/index.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+In order to follow this guide, you will need to install Python in your local machine.
+
+[python]: https://www.python.org/
+[falcon]: https://falconframework.org/
+
+## Objectives
+
+1. Create a Python app that uses falcon as a framework.
+2. Run the app locally.
+3. Deploy the Python app to Google App Engine standard environment.
```
Comment (devlance) [inline tutorials/appengine-python-falcon/index.md]: The original is correct. Standard environment isn't part of the product name. @chenopis See our internal guidelines.

Comment (chenopis) [inline tutorials/appengine-python-falcon/index.md]: Ok, I withdraw this comment then. I guess I just disagree w/ how we name things. :(

File: tutorials/appengine-python-falcon/index.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+2. Visit `http://[YOUR_PROJECT_ID].appspot.com` to see the deployed app.
+
+    Replace `[YOUR_PROJECT_ID]` with your Google Cloud Platform project ID.
+
+3. Run the following command to view your app
+
+    gcloud app browse
+
+[requirements]: https://github.com/GoogleCloudPlatform/community/tree/master/tutorials/appengine-python-falcon/requirements.txt
```
Comment (jmdobry) [inline tutorials/appengine-python-falcon/index.md]: This link needs to be viewed from `cloud.google.com/community/tutorials/appengine-python-falcon`, a relative link won't be able to point back to github.com from there.

Comment (chenopis) [inline tutorials/appengine-python-falcon/index.md]: Ah, ok. So GitHub is still supposed to host the tutorial example files even though the tutorial will be on CGC. Got it.

**Comment by jmdobry** [comment]
Published at https://cloud.google.com/community/tutorials/appengine-python-falcon

**Pull request merged**